src/aiida/tools/dumping/processes.py

```diff
@@ -0,0 +1,498 @@
+        # `dump` function called by `verdi`, then I need to dump for the `CalculationNode` here already, as well.
+        if isinstance(process_node, CalculationNode):
+            self._dump_calculation(
+                calculation_node=process_node,
+                output_path=output_path,
+                io_dump_paths=io_dump_paths,
+            )
+            # Don't actually to call `self.dump_node_yaml` here, as this is done inside `self._dump_calculation`
+
+        elif isinstance(process_node, WorkflowNode):
+            # Can call `self.dump_node_yaml` here before the actual dump, as file dumping happens in child directories
+            self.dump_node_yaml(process_node=process_node, output_path=output_path)
+
+            called_links = process_node.base.links.get_outgoing(
+                link_type=(LinkType.CALL_CALC, LinkType.CALL_WORK)
+            ).all()
+
+            sorted_called_links = sorted(called_links, key=lambda link_triple: link_triple.node.ctime)
+
+            for index, link_triple in enumerate(sorted_called_links, start=1):
+                child_node = link_triple.node
+                child_label = self.generate_child_node_label(index=index, link_triple=link_triple)
+                child_output_path = output_path.resolve() / child_label
+
+                # Recursive function call for `WorkFlowNode``
+                if isinstance(child_node, WorkflowNode):
+                    self.dump(
+                        process_node=child_node,
+                        output_path=child_output_path,
+                        io_dump_paths=io_dump_paths,
+                    )
+
+                # Once a `CalculationNode` as child reached, dump it
+                elif isinstance(child_node, CalculationNode):
+                    self._dump_calculation(
+                        calculation_node=child_node,
+                        output_path=child_output_path,
+                        io_dump_paths=io_dump_paths,
+                    )
+
+    def _dump_calculation(
+        self,
+        calculation_node: CalculationNode,
+        output_path: Path | None,
+        io_dump_paths: list | None = None,
+    ) -> None:
+        """Dump the contents of a `CalculationNode` to a specified output path.
+
+        :param calculation_node: The `CalculationNode` to be dumped.
+        :param output_path: The path where the files will be dumped.
+        :param io_dump_paths: Subdirectories created for the `CalculationNode`.
+            Default: ['raw_inputs', 'extra_inputs', 'raw_outputs']
+        """
+
+        if output_path is None:
+            output_path = self.generate_default_dump_path(process_node=calculation_node)
+
+        try:
+            self.validate_make_dump_path(validate_path=output_path)
+        except:
+            # raise same exception here to communicate it outwards
+            raise
```

Comment:
I think this was to be able to capture it in `cmd_process` and provide a message to the user via `echo.echo_`, rather than using the logger here. Will fix it!